File: tourism/src/main/java/com/tourism/Tourist.java
package com.tourism;

/**
 * Клас для представлення туриста
 */
public class Tourist extends TravelSubject {
    private Booking booking;

    public Tourist(String name) {
        super(name);
    }

    @Override
    public void showRole() {
        System.out.println(name + " — Турист.");
    }

    @Override
    public void makeReservation(String roomType) {
        System.out.println(name + " бронює номер: " + roomType);
    }

    @Override
    public void cancelReservation() {
        System.out.println(name + " скасовує бронювання.");
    }

    public void setBooking(Booking booking) {  // Агрегація
        this.booking = booking;
    }

    public void changeDates(String newDates) {
        System.out.println(name + " змінює дати: " + newDates);
    }
}

File: tourism-python/src/tourism/tourist.py
from tourism.subjects import TravelSubject


class Tourist(TravelSubject):
    """
    Клас для представлення туриста
    """

    def __init__(self, name: str):
        super().__init__(name)
        self.booking = None

    def show_role(self):
        print(f"{self.name} — Турист.")

    def make_reservation(self, room_type: str):
        print(f"{self.name} бронює номер: {room_type}")

    def cancel_reservation(self):
        print(f"{self.name} скасовує бронювання.")

    def set_booking(self, booking):
        self.booking = booking

    def change_dates(self, new_dates: str):
        print(f"{self.name} змінює дати: {new_dates}")
